Testing code in docs does not compile

Hi,
(Similar to issue #611) I am testing some filters using trying to use the examples from the docs but running into a compiler error:

```
use warp::Filter;

fn sum() -> impl Filter<Extract = (u32,), Error = warp::Rejection> + Copy {
    warp::path::param()
        .and(warp::path::param())
        .map(|x: u32, y: u32| x + y)
}

fn math() -> impl Filter<Extract = (String,), Error = warp::Rejection> + Copy {
    warp::post().and(sum()).map(|z: u32| format!("Sum = {}", z))
}

#[cfg(test)]
mod tests {
    use super::*;

    #[tokio::test]
    async fn test_sum() {
        let filter = sum();

        // Execute `sum` and get the `Extract` back.
        let value = warp::test::request()
            .path("/1/2")
            .filter(&filter)
            .await
            .unwrap();
        assert_eq!(value, 3);

        // Or simply test if a request matches (doesn't reject).
        assert!(!warp::test::request().path("/1/-5").matches(&filter));
    }
}
```

```
error[E0600]: cannot apply unary operator `!` to type `impl warp::Future`
  --> src/main.rs:30:17
   |
30 |         assert!(!warp::test::request().path("/1/-5").matches(&filter));
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot apply unary operator `!`
```

All that is missing here is to call .await on the Future within the assert! macro. I can submit a PR for this.

--- a/src/test.rs
+++ b/src/test.rs
@@ -51,6 +51,7 @@
 //!         !warp::test::request()
 //!             .path("/1/-5")
 //!             .matches(&filter)
+//!             .await
 //!     );
 //! }
 //! ```

